File: event-consumer/src/test/java/nl/haploid/octowight/kafka/KafkaConsumerFactoryIT.java
package nl.haploid.octowight.kafka;

import kafka.consumer.KafkaStream;
import kafka.javaapi.consumer.ConsumerConnector;
import nl.haploid.octowight.AbstractIT;
import nl.haploid.octowight.TestData;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import static org.junit.Assert.assertNotNull;

public class KafkaConsumerFactoryIT extends AbstractIT {

	@Autowired
	private KafkaConsumerFactory service;

	@Test
	public void testCreateKafkaConsumer() {
		final ConsumerConnector kafkaConsumer = service.createKafkaConsumer();
		assertNotNull(kafkaConsumer);
	}

	@Test
	public void testCreateStream() {
		final String topic = TestData.topic();
		final ConsumerConnector kafkaConsumer = service.createKafkaConsumer();
		final KafkaStream<byte[], byte[]> stream = service.createStream(kafkaConsumer, topic);
		assertNotNull(stream);
	}
}

File: event-consumer/src/test/scala/nl/haploid/octowight/kafka/KafkaConsumerFactoryIT.scala
package nl.haploid.octowight.kafka

import nl.haploid.octowight.{AbstractIT, TestData}
import org.springframework.beans.factory.annotation.Autowired

class KafkaConsumerFactoryIT extends AbstractIT {
  @Autowired private val kafkaConsumerFactory: KafkaConsumerFactory = null

  "Kafka consumer factory" should "build a consumer" in {
    val kafkaConsumer = kafkaConsumerFactory.createKafkaConsumer
    kafkaConsumer should not be null
  }

  "Kafka consumer factory" should "build a stream" in {
    val topic = TestData.topic
    val kafkaConsumer = kafkaConsumerFactory.createKafkaConsumer
    val stream = kafkaConsumerFactory.createStream(kafkaConsumer, topic)
    stream should not be null
  }
}
